src/Api/Controllers/CiphersController.cs

```diff
@@ -398,6 +418,24 @@ public async Task PutDeleteMany([FromBody]CipherBulkRestoreRequestModel model)
             await _cipherService.SoftDeleteManyAsync(model.Ids.Select(i => new Guid(i)), userId);
         }
 
+        [HttpPut("delete-admin")]
+        public async Task PutDeleteManyAdmin([FromBody]CipherBulkRestoreRequestModel model)
+        {
+            if (!_globalSettings.SelfHosted && model.Ids.Count() > 500)
+            {
+                throw new BadRequestException("You can only restore up to 500 items at a time.");
+            }
+
+            var userId = _userService.GetProperUserId(User).Value;
+            var ciphers = await _cipherRepository.GetManyByIdAsync(model.Ids.Select(i => new Guid(i)));
+            ciphers = ciphers
+                .Where(cipher => ciphers != null
+                        && cipher.OrganizationId.HasValue
+                        && _currentContext.OrganizationAdmin(cipher.OrganizationId.Value)).ToList();
+
+            await _cipherService.SoftDeleteManyAsync(ciphers.Select(cipher => cipher.Id), userId);
```

Comment:
Two comments here:

1. There is business logic here that should probably be extracted to the service.
2. I don't think your call to `SoftDeleteManyAsync` will work as expected since `_cipherRepository.GetManyByUserIdAsync(deletingUserId)` does not return ciphers to an admin unless they are assigned to a collection that can access them. This is the difference between an org admin and regular user. A org admin should be able to delete any cipher that is part of their organization.

Reply:
I updated this to optionally carry the organizationId up from the client, and pull the list of available ciphers from that instead if coming from an admin. Let me know your thoughts.